src/state_stack.cpp

```diff
@@ -0,0 +1,159 @@
+                [&](issues::ExtraneousElement &i) { std::cout << "There was an extre element in the module definition in module " << i.mod << " at " << i.in << '\n';  },
+                [&](issues::RequiredModuleNotFound &i) { std::cout << "Module " << i.requiree << " requires the module " << i.required << ", but it was not found\n"; any_issues_critical = true; },
+                [&](issues::InvalidFile &i) { std::cout << "Module " << i.what_module << " tried to load the file " << i.filepath << ", but it could not be loaded\n"; any_issues_critical = true; },
+                [&](issues::MissingField &i) { std::cout << "Module " << i.what_module << " provided a definition of \"" << i.what_def << "\", but it was missing the field " << i.fieldname << "\n"; any_issues_critical = true; },
+                [&](issues::InvalidKey &i) { std::cout << "Module " << i.what_module << " provided an invalid key when defining " << i.what_def << '\n'; any_issues_critical = true; }
+        }, issue);
+    };
+
+    for(const auto& issue : moduleLoader.LoadModules(module_load_candidates, inputMgr, resourceStore)) {
+        deal_with_an_issue(issue);
+    }
+    if (any_issues_critical) {
+        exit(1);
+    }
+
+    for(const auto& issue : resourceStore.LoadModuleResources(moduleLoader)) {
+        deal_with_an_issue(issue);
+    }
+    if (any_issues_critical) {
+        exit(1);
+    }
+}
+
+void State_Stack::perform_queued_actions()
+{
+	while (!actions_queue.empty())
+	{
+		ACTIONS id = actions_queue.front();
+		actions_queue.pop();
+
+		switch (id)
+		{
+		case(ACTIONS::PUSH):
+		{
+			auto state = states_queue.front();
+			states_queue.pop();
+			states_stack.emplace_back(factories.find(state)->second(*this), state);
+			break;
+		}
+		case(ACTIONS::POP):
+		{
+			if (!states_stack.empty())
+				states_stack.pop_back();
+			break;
+		}
+		case(ACTIONS::CLEAR):
+		{
+			while (!states_stack.empty())
+				states_stack.pop_back();
+			break;
+		}
+		}
+	}
+}
+
+void State_Stack::handle_events()
+{
+	if (WindowShouldClose())
+		this->request_clear();
+
+	if (IsWindowResized())
+		for (const auto& state : states_stack)
+			state.first->adjust_to_window();
+
+	states_stack.back().first->handle_events();
+}
+
+void State_Stack::update()
+{
+	while (time_passed > time_per_frame)
+	{
+		time_passed -= time_per_frame;
+		states_stack.back().first->update(time_per_frame);
+	}
+}
+
+void State_Stack::draw()
+{
+	BeginDrawing();
+	ClearBackground(WHITE);
+
+	states_stack.back().first->render();
+
+	EndDrawing();
+}
+
+void State_Stack::request_push(STATES state_id)
+{
+	actions_queue.emplace(ACTIONS::PUSH);
+	states_queue.emplace(state_id);
+}
+
+void State_Stack::request_pop()
+{
+	actions_queue.emplace(ACTIONS::POP);
+}
+
+void State_Stack::request_clear()
+{
+	actions_queue.emplace(ACTIONS::CLEAR);
+}
+
+void State_Stack::run()
+{
+	while (!states_stack.empty())
+	{
+		auto t1 = std::chrono::high_resolution_clock::now();

```

Comment:
high_resolution_clock jest useless. użyj steady_clock

Reply:
thank you for your opinion